src/nameserver/name_server_impl.h

```diff
@@ -467,135 +450,26 @@ class NameServerImpl : public NameServer {
 
     int UpdateEndpointTableAlive(const std::string& endpoint, bool is_alive);
 
-    std::shared_ptr<Task> CreateMakeSnapshotTask(const std::string& endpoint, uint64_t op_index,
-                                                 ::openmldb::api::OPType op_type, uint32_t tid, uint32_t pid,
-                                                 uint64_t end_offset);
-
-    std::shared_ptr<Task> CreatePauseSnapshotTask(const std::string& endpoint, uint64_t op_index,
-                                                  ::openmldb::api::OPType op_type, uint32_t tid, uint32_t pid);
-
-    std::shared_ptr<Task> CreateRecoverSnapshotTask(const std::string& endpoint, uint64_t op_index,
-                                                    ::openmldb::api::OPType op_type, uint32_t tid, uint32_t pid);
-
-    std::shared_ptr<Task> CreateSendSnapshotTask(const std::string& endpoint, uint64_t op_index,
-                                                 ::openmldb::api::OPType op_type, uint32_t tid, uint32_t remote_tid,
-                                                 uint32_t pid, const std::string& des_endpoint);
-
-    std::shared_ptr<Task> CreateLoadTableTask(const std::string& endpoint, uint64_t op_index,
-                                              ::openmldb::api::OPType op_type, const std::string& name, uint32_t tid,
-                                              uint32_t pid, uint32_t seg_cnt, bool is_leader,
-                                              ::openmldb::common::StorageMode storage_mode);
+    std::shared_ptr<Task> CreateTask(const std::shared_ptr<TaskMeta>& task_meta);
```

Comment:
can simplified with a template wrapper,  checkout https://github.com/4paradigm/OpenMLDB/blob/0920bf256d10a1c578a1da7ed7bfdb17d23efab7/hybridse/include/node/node_manager.h#L64-L68

Reply:
`CreateTask` function is not only creating Task object, but also doing some init works like binding different execute function. 
Absolutely, using template can simplify the code of calling `CreateTask` instead of using shared_ptr

Reply:
You can make a template wrapper over the current `CreateTask` in header, no need to move impls into header

Reply:
will update in #3288 later